### PR TITLE
chore(dp): Add alembic-autogen-check to CI

### PR DESCRIPTION
**Pull request by sklgromek**
Signed-off-by: Tomasz Gromowski <tomasz@freedomfi.com>

## Summary
Add the alembic autogen check step to CI to check if the model was changed, and the database will need migration 

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: .github/workflows/dp-workflow.yml [hunk 1]
```diff
@@ -333,6 +333,48 @@ jobs:
...
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install alembic-autogen-check
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+
+      - name: Database migration check
+        working-directory: "${{ github.workspace }}/dp/cloud/python/magma/db_service/migrations"
+        run: |
+          alembic upgrade heads
```
Comment (xbend) [inline .github/workflows/dp-workflow.yml]: change `heads` to `head`.
We do not want to apply multiple heads as we want the revision history to be linear.

Comment (sklgromek) [reply .github/workflows/dp-workflow.yml]: done

**Review by jkmar — APPROVED** [review]

**Comment by jkmar** [comment]
@sklgromek Please rebase with master, migration fix has just been merged there

**Comment by sebathomas** [comment]
Please add a description to the table after this check is merged:
https://github.com/magma/magma/wiki/Contributing-Code

**Review by xbend — APPROVED** [review]

**Comment by xbend** [comment]
> Please add a description to the table after this check is merged: https://github.com/magma/magma/wiki/Contributing-Code

@sebathomas 
Will do. We created [an issue](https://app.zenhub.com/workspaces/magma--domain-proxy-development-6064cb9d84ca0600135548c9/issues/magma/domain-proxy/363) to track this addition.

**Comment by xbend** [comment]
@119Vik @quentinDERORY 
Change is ready for merge and approval from CI side.

**Review by 119Vik — APPROVED** [review]

**Pull request merged**